pylint/checkers/variables.py

```diff
@@ -606,9 +606,25 @@ def leave_setcomp(self, _):
         # do not check for not used locals here
         self._to_consume.pop()
 
+    def _consume_in_scope(self, name, target_scope):
+        """move a name from to_consume to consumed in given scope
+        """
+        if not name:
+            return
+        for to_consume, consumed, scope in self._to_consume:
+            if scope == target_scope:
+                if name in to_consume:
+                    consumed[name] = to_consume[name]
+                    del to_consume[name]
+
     def visit_functiondef(self, node):
         """visit function: update consumption analysis variable and check locals
         """
+        for arg_annotation in node.args.annotations:
+            if isinstance(arg_annotation, astroid.Const):
```

Comment:
I'd limit this check for forward references only, so check strings only if enclosing scope actually defines this name. All other string usages are violations.